kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/AxonSerializers.kt

```diff
@@ -0,0 +1,333 @@
+            }
+        }
+        MultiSourceTrackingToken(
+            trackingTokens ?: throw SerializationException("Element 'trackingTokens' is missing"),
+        )
+    }
+
+    override fun serialize(encoder: Encoder, value: MultiSourceTrackingToken) = encoder.encodeStructure(descriptor) {
+        encodeSerializableElement(descriptor, 0, mapSerializer, value.trackingTokens)
+    }
+}
+
+object MergedTrackingTokenSerializer : KSerializer<MergedTrackingToken> {
+
+    override val descriptor = buildClassSerialDescriptor(MergedTrackingToken::class.java.name) {
+        element<TrackingToken>("lowerSegmentToken")
+        element<TrackingToken>("upperSegmentToken")
+    }
+
+    override fun deserialize(decoder: Decoder) = decoder.decodeStructure(descriptor) {
+        var lowerSegmentToken: TrackingToken? = null
+        var upperSegmentToken: TrackingToken? = null
+        while (true) {
+            val index = decodeElementIndex(descriptor)
+            if (index == CompositeDecoder.DECODE_DONE) break
+            when (index) {
+                0 -> lowerSegmentToken = decodeSerializableElement(descriptor, index, trackingTokenSerializer)
+                1 -> upperSegmentToken = decodeSerializableElement(descriptor, index, trackingTokenSerializer)
+            }
+        }
+        MergedTrackingToken(
+            lowerSegmentToken ?: throw SerializationException("Element 'lowerSegmentToken' is missing"),
+            upperSegmentToken ?: throw SerializationException("Element 'upperSegmentToken' is missing"),
+        )
+    }
+
+    override fun serialize(encoder: Encoder, value: MergedTrackingToken) = encoder.encodeStructure(descriptor) {
+        encodeSerializableElement(descriptor, 0, trackingTokenSerializer, value.lowerSegmentToken())
+        encodeSerializableElement(descriptor, 1, trackingTokenSerializer, value.upperSegmentToken())
+    }
+}
+
+object ReplayTokenSerializer : KSerializer<ReplayToken> {
+
+    override val descriptor = buildClassSerialDescriptor(ReplayToken::class.java.name) {
+        element<TrackingToken>("tokenAtReset")
+        element<TrackingToken>("currentToken")
+    }
+
+    override fun deserialize(decoder: Decoder) = decoder.decodeStructure(descriptor) {
+        var tokenAtReset: TrackingToken? = null
+        var currentToken: TrackingToken? = null
+        while (true) {
+            val index = decodeElementIndex(descriptor)
+            if (index == CompositeDecoder.DECODE_DONE) break
+            when (index) {
+                0 -> tokenAtReset = decodeSerializableElement(descriptor, index, trackingTokenSerializer)
+                1 -> currentToken = decodeSerializableElement(descriptor, index, trackingTokenSerializer)
+            }
+        }
+        ReplayToken(
+            tokenAtReset ?: throw SerializationException("Element 'tokenAtReset' is missing"),
+            currentToken,
+        )
+    }
+
+    @OptIn(ExperimentalSerializationApi::class)
```

Comment:
It would be nice if we could refrain from using experimental api's in src. Although small, there is a risk these will change and not be compatible with future versions of Kotlin.

Reply:
This was actually no longer needed, so I removed the opt-in